protocol-units/execution/suzuka/fin-executor/src/executor.rs

```diff
@@ -0,0 +1,233 @@
+use aptos_db::AptosDB;
+use aptos_executor_types::{state_checkpoint_output::StateCheckpointOutput, BlockExecutorTrait};
+use aptos_mempool::core_mempool::CoreMempool;
+use aptos_storage_interface::DbReaderWriter;
+use aptos_types::{
+	block_executor::config::BlockExecutorConfigFromOnchain,
+	transaction::{
+		Transaction, WriteSetPayload,
+	},
+	validator_signer::ValidatorSigner,
+	block_executor::partitioner::ExecutableBlock
+};
+use aptos_vm::AptosVM;
+use std::{path::PathBuf, sync::{Arc, RwLock}};
+use aptos_config::config::NodeConfig;
+use aptos_executor::{
+	block_executor::BlockExecutor,
+	db_bootstrapper::{generate_waypoint, maybe_bootstrap},
+};
+
+/// The state of `movement-network` execution can exist in three states,
+/// `Dynamic`, `Optimistic`, and `Final`. The `Dynamic` state is the state.
+pub enum FinalityState {
+	/// The dynamic state that is subject to change and is not
+	/// yet finalized. It is the state that is derived from the blocks
+	/// received before any finality is reached and simply represents a
+	/// local application of the fork-choice rule (longest chain)
+	/// of the gossipped blocks.
+	Dynamic,
+	/// The optimistic state that is derived from the blocks received after DA finality.
+	/// It is the state that is derived from the blocks that have been finalized by the DA.
+	Optimistic,
+	/// The final state that is derived from the blocks received after the finality is reached.
+	Final,
+}
+
+/// The current state of the executor and its execution of blocks.
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum ExecutorState {
+	/// The executor is idle and waiting for a block to be executed.
+	Idle,
+	/// The block is executed in a speculative manner and its effects held in memory.
+	Speculate,
+	/// The network agrees on the block.
+	Consensus,
+	/// The block is committed to the state, at this point
+	/// fork choices must be resolved otherwise the commitment and subsequent execution will fail.
+	Commit,
+}
```

Comment:
Let's remove this enum, its dead code now. And all its doc comments

Reply:
@l-monninger agree?

Reply:
Yeah, I honestly thought I had already dropped this.